craft_application/launchpad/models/recipe.py

```diff
@@ -244,13 +246,27 @@ def new(  # noqa: PLR0913
                 "An auto-build archive may only be provided if auto-build is enabled.",
             )
 
-        snap_entry = lp.lp.snaps.new(
-            name=name,
-            owner=util.get_person_link(owner),
-            store_upload=bool(store_name),
-            auto_build=auto_build,
-            **kwargs,
-        )
+        snap_entry = None
+
+        for attempt in range(_ATTEMPT_COUNT):
+            try:
+                snap_entry = lp.lp.snaps.new(
```

Comment:
emit a debug message here, something like f"Trying to Create project ({attempt}/{_ATTEMPT_COUNT}"